### PR TITLE
:sparkles: Update keras image-classification pipeline

### PR DESCRIPTION
**Pull request by nateraw**

### CODE REVIEW

File: api-inference-community/docker_images/keras/app/pipelines/image_classification.py [hunk 1]
```diff
@@ -45,8 +66,9 @@ def __call__(self, inputs: "Image.Image") -> List[Dict[str, Any]]:
         expected_input_size = self.model.input_shape
         if expected_input_size[-1] == 1:  # Single channel, we assume grayscale
             inputs = inputs.convert("L")
+
         target_size = (expected_input_size[1], expected_input_size[2])
-        img = inputs.resize(target_size)
+        img = inputs.resize(target_size, self.interpolation)
```
Comment (nateraw) [inline api-inference-community/docker_images/keras/app/pipelines/image_classification.py:71]: Not sure if interpolation is needed, but its in [Keras's `load_img` function](https://github.com/keras-team/keras-preprocessing/blob/6701f27afa62712b34a17d4b0ff879156b0c7937/keras_preprocessing/image/utils.py#L76-L172), so I decided to add it.

File: api-inference-community/docker_images/keras/requirements.txt [hunk 2]
```diff
@@ -1,5 +1,8 @@
 starlette==0.14.2
 api-inference-community==0.0.8
 numpy>=1.17
-tensorflow>=2.3
-huggingface_hub===0.0.13
\ No newline at end of file
+tensorflow==2.6.0
+# huggingface_hub===0.0.13
```
Comment (nateraw) [inline api-inference-community/docker_images/keras/requirements.txt:5]: @LysandreJik  Can we perhaps do dev release before we merge this so we don't have to have git install here?

File: api-inference-community/docker_images/keras/app/pipelines/image_classification.py [hunk 3]
```diff
@@ -1,35 +1,51 @@
...
+
+        # Reload Keras SavedModel
+        self.model = from_pretrained_keras(model_id)
 
         # Handle binary classification with a single output unit.
         self.single_output_unit = self.model.output_shape[1] == 1
+        self.num_labels = 2 if self.single_output_unit else self.model.output_shape[1]
 
         # Config is required to know the mapping to label.
         config_file = cached_download(hf_hub_url(model_id, filename=CONFIG_FILENAME))
```
Comment (osanseviero) [inline api-inference-community/docker_images/keras/app/pipelines/image_classification.py:36]:
```suggestion
        config_file = hf_hub_download(model_id, filename=CONFIG_FILENAME))
```

File: api-inference-community/docker_images/keras/app/pipelines/image_classification.py [hunk 4]
```diff
@@ -1,35 +1,51 @@
...
         with open(config_file) as config:
             config = json.load(config)
-        self.id2label = config["id2label"]
+
+        self.labels = config.get("labels", None)
```
Comment (osanseviero) [inline api-inference-community/docker_images/keras/app/pipelines/image_classification.py:40]: Any reason for this change? I think `id2label` is the usual approach in `transformers` and would prefer to keep consistent unless there's a strong reason.

File: api-inference-community/docker_images/keras/app/pipelines/image_classification.py (hunk 3 above)
Comment (osanseviero) [inline api-inference-community/docker_images/keras/app/pipelines/image_classification.py:36]: qq: is this automatically created at the moment?

File: api-inference-community/docker_images/keras/app/pipelines/image_classification.py [hunk 5]
```diff
@@ -63,4 +80,4 @@ def __call__(self, inputs: "Image.Image") -> List[Dict[str, Any]]:
                 {"label": str(self.id2label[str(i)]), "score": float(score)}
                 for i, score in enumerate(predictions[0])
             ]
-        return sorted(labels, key=lambda tup: tup["score"], reverse=True)
+        return sorted(labels, key=lambda tup: tup["score"], reverse=True)[: self.top_k]
```
Comment (osanseviero) [inline api-inference-community/docker_images/keras/app/pipelines/image_classification.py:83]: Great idea to add top_k

File: api-inference-community/tests/test_dockers.py [hunk 6]
```diff
@@ -160,8 +160,13 @@ def test_timm(self):
         self.framework_invalid_test("timm")
```
Comment (osanseviero) [inline api-inference-community/tests/test_dockers.py:160]: Not needed, but could you add a forked version of https://github.com/huggingface/huggingface_hub/blob/main/.github/workflows/python-api-allennlp.yaml that is modified for `keras`? We'll merge them in the future but I haven't gotten to investigating it.

File: api-inference-community/tests/test_dockers.py [hunk 7]
```diff
@@ -160,8 +160,13 @@ def test_timm(self):
...
 
     def test_keras(self):
+        # Single Output Unit
         self.framework_docker_test(
-            "keras", "image-classification", "osanseviero/keras-dog-or-cat"
+            "keras", "image-classification", "nateraw/keras-cats-vs-dogs"
+        )
+        # Multiple Output Units
```
Comment (osanseviero) [inline api-inference-community/tests/test_dockers.py:167]: nit: maybe also mention if it's grayscale or rgb

**Review by osanseviero — APPROVED** [review]
Looks great! Thanks for adding this!

File: api-inference-community/docker_images/keras/app/pipelines/image_classification.py (hunk 3 above)
Comment (nateraw) [reply api-inference-community/docker_images/keras/app/pipelines/image_classification.py]: you mean `self.model.output_shape`? That's created automatically by Keras.

Comment (osanseviero) [reply api-inference-community/docker_images/keras/app/pipelines/image_classification.py]: Sorry, I meant the config file

Comment (nateraw) [reply api-inference-community/docker_images/keras/app/pipelines/image_classification.py]: it does if you do `push_to_hub_keras(model, 'some-model', config=dict(id2label={'0': 'asdf', '1': 'blah'}))`

File: api-inference-community/docker_images/keras/requirements.txt (hunk 2 above)
Comment (LysandreJik) [reply api-inference-community/docker_images/keras/requirements.txt:5]: Yes! Will release in a minute.

**Pull request merged**